R/benchmark-results.R

```diff
@@ -27,13 +36,30 @@ benchmark_results <- function(run_id = NULL, batch_id = NULL, run_reason = NULL,
     req <- req_url_query(req, run_reason = run_reason)
   }
 
+  if (!is.null(days)) {
+    ## assert that days is a number
+    stopifnot("days must be numeric" = is.numeric(days))
+    req <- req_url_query(req, days = days)
+  }
+
+  if (!is.null(limit) && !is.null(run_reason)) {
+    ## assert that limit is a number
+    stopifnot("limit must be numeric" = is.numeric(limit))
+    req <- req_url_query(req, limit = limit)
+  }
```

Comment:
We should maybe have a test when days/limits is set to `0` or partial values (1.5 etc) it not null & is.numeric is nature 

Not sure if the main API handles that

Reply:
Oh good catch. I think for 0 it is ok. The api behaves as one would expect. For `limit = 0` it returns nothing. And then for `days = 0` it does not look back any days - rather it just grabs values from today. The slight nuance is that if there happens to be no benchmark results from yesterday then `days = 1` will look identical to `days = 0`. 

But non-integers did need to be handled. I am opting to message and coerce because it would be annoying to provide `limit = 8` (instead of `limit = 8L`) and have the function error out.